src/lib/Types/Core.hs

```diff
@@ -1663,6 +1683,58 @@ instance SubstE AtomSubstVal PiType
 deriving instance Show (PiType n)
 deriving via WrapE PiType n instance Generic (PiType n)
 
+instance GenericB RolePiBinder where
+  type RepB RolePiBinder = BinderP AtomNameC (PairE Type (LiftE (Arrow, ParamRole)))
+  fromB (RolePiBinder b ty arr role) = b :> PairE ty (LiftE (arr, role))
+  toB   (b :> PairE ty (LiftE (arr, role))) = RolePiBinder b ty arr role
```

Comment:
Please always mark generic methods INLINE

Reply:
Thanks for catching these. I'm trying to build the habit but I'm not there yet.